### PR TITLE
Add optional client for control server to launcher

### PR DESCRIPTION
**Pull request by marpaia**
So far, this PR:

- Adds an off-by-default killswitch for the control server

### CODE REVIEW

File: cmd/launcher/options.go [hunk 1]
```diff
@@ -240,6 +255,7 @@ func shortUsage() {
 	printOpt("osqueryd_path")
 	fmt.Fprintf(os.Stderr, "\n")
 	printOpt("autoupdate")
+	printOpt("control")
```
Comment (NickTitle) [inline cmd/launcher/options.go]: missed a newline

File: tools/packaging/packaging.go [hunk 2]
```diff
@@ -40,18 +40,20 @@ func CreatePackages(
...
+	control bool,
+	controlHostname string,
 	identifier string,
 	omitSecret bool,
 	systemd bool,
 	certPins,
 	rootPEM string,
 ) (*PackagePaths, error) {
-	macPkgDestinationPath, err := CreateMacPackage(packageVersion, osqueryVersion, hostname, secret, macPackageSigningKey, insecure, insecureGrpc, autoupdate, updateChannel, identifier, omitSecret, certPins, rootPEM)
+	macPkgDestinationPath, err := CreateMacPackage(packageVersion, osqueryVersion, hostname, secret, macPackageSigningKey, insecure, insecureGrpc, autoupdate, updateChannel, control, controlHostname, identifier, omitSecret, certPins, rootPEM)
```
Comment (NickTitle) [inline tools/packaging/packaging.go]: a PackageParams struct is starting to look pretty good :P

File: cmd/launcher/options.go (hunk 1 above)
Comment (marpaia) [reply cmd/launcher/options.go]: it's intentional, i wanted to put the kill switches together

File: tools/packaging/packaging.go (hunk 2 above)
Comment (marpaia) [reply tools/packaging/packaging.go]: yeah, one day...

File: cmd/package-builder/package-builder.go [hunk 3]
```diff
@@ -132,7 +142,7 @@ func runMake(args []string) error {
 	}
 
 	currentVersion := version.Version().Version
-	paths, err := packaging.CreatePackages(currentVersion, osqueryVersion, *flHostname, *flEnrollSecret, macPackageSigningKey, *flInsecure, *flInsecureGrpc, *flAutoupdate, *flUpdateChannel, *flIdentifier, *flOmitSecret, *flSystemd, *flCertPins, *flRootPEM)
+	paths, err := packaging.CreatePackages(currentVersion, osqueryVersion, *flHostname, *flEnrollSecret, macPackageSigningKey, *flInsecure, *flInsecureGrpc, *flAutoupdate, *flUpdateChannel, *flControl, *flControlHostname, *flIdentifier, *flOmitSecret, *flSystemd, *flCertPins, *flRootPEM)
```
Comment (groob) [inline cmd/package-builder/package-builder.go]: let's refactor this to wrap the options one per line. Maybe it's time for a struct?

Comment (marpaia) [reply cmd/package-builder/package-builder.go]: i think refactoring this to a struct, or better functional options, is out of scope for this PR, but i'll consider revisiting it when there's more time.

File: control/control.go [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+	}, nil
+}
+
+func (c *Client) Start(ctx context.Context) {
+	ctx, c.cancel = context.WithCancel(ctx)
+	getShellsTicker := time.NewTicker(c.getShellsInterval)
+	for {
+		select {
+		case <-ctx.Done():
+			return
```
Comment (groob) [inline control/control.go]: log ctx.Err() before returning

**Comment by marpaia** [comment]
Closing this. @loganmac will re-open.

**Pull request closed without merging**